### DISCUSSION TITLE
Testing out facts that are going to be depreciated

### DISCUSSION THREAD

**Discussion by patrickmayes** [puppetlabs/pdk#1022] [Feature Request]
Facts are at time depreciated, if a module has a fact listed that will be depreciated it would be nice to see a warning stating that the fact is going to be depreciated in a future version of puppet.

**Comment by bmjen** [comment]
Thanks @patrickmayes !

The deprecation of facts should be reporting a warning from puppet itself before being fully deprecated. I will check with the core team on whether this is accurate. 

The idea of improving community tools to detect potentially deprecated facts sounds like a good idea and something we should look into. It's also something that can be eligible for "Needs RFC" which opens the door for a contributor to submit a proposed implementation and subsequently implement the contribution.
